File: e-dashboard-ui/src/ClientApp/src/views/Diagram/DiagramLib/Shapes/TextStyle.ts

import { ITextStyle } from './interfaces';

type CanvasTextAlign = "left" | "right" | "center" | "start" | "end";
type CanvasTextBaseline= "top" | "hanging" | "middle" | "alphabetic" | "ideographic" | "bottom";

export class TextStyle implements ITextStyle {
    public opacity: number;
    public strokeWidth: number;
    public strokeStyle: string;
    public fillStyle: string;
    public fontFamily: string;
    public fontSize: number;
    public fontStyle: string;
    public textAlign: CanvasTextAlign;
    public textBaseline: CanvasTextBaseline;
    public textAnchor: string;

    constructor() {
        this.opacity = 1;
        this.strokeWidth = 1;
        this.strokeStyle = "#000000";
        this.fillStyle = "#101010";
        this.fontFamily = "Vazirmatn";
        this.fontSize = 8;
        this.fontStyle = "normal"; //oblique normal, italic, or bold
        this.textAlign = 'end';// start, end, left, right, center
        this.textBaseline = 'middle';//top hanging middle alphabetic ideographic bottom
        this.textAnchor = 'middle';
    }
}

File: e-dashboard-api/src/Domain/Diagram/Shapes/TextStyle.cs

namespace EnergyDashboard.Domain.Diagram.Shapes;

public class TextStyle : Style
{
    public TextStyle()
    {
        this.Opacity = 1;
        this.StrokeWidth = 1;
        this.StrokeStyle = "#000000";
        this.FillStyle = "#101010";
        this.FontFamily = "Vazir FD";
        this.FontSize = 8;
        this.FontStyle = "normal"; //oblique normal, italic, or bold
        this.TextAlign = "end"; // start, end, left, right, center
        this.TextBaseline = "middle"; //top hanging middle alphabetic ideographic bottom
        this.TextAnchor = "middle";
    }
    public string FontFamily { get; set; }
    public int FontSize { get; set; }
    public string FontStyle { get; set; }
    public string TextAlign { get; set; }
    public string TextBaseline { get; set; }
    public string TextAnchor { get; set; }
}
